servicetalk-opentelemetry-http/docs/modules/ROOT/pages/index.adoc

```diff
@@ -0,0 +1,220 @@
+
+=== Client Filters
+
+When using multiple filters on clients, place the OpenTelemetry filter appropriately in the chain:
+
+[source,java]
+----
+HttpClient client = HttpClients.forSingleAddress("example.com", 80)
+    // Logging or other observability filters before tracing
+    .appendClientFilter(loggingFilter)
+
+    // OpenTelemetry filter for span creation and context propagation
+    .appendClientFilter(new OpenTelemetryHttpRequesterFilter.Builder()
+        .componentName("my-client")
+        .build())
+
+    // Retry and other resilience filters after tracing
+    .appendClientFilter(retryFilter)
+    .build();
+----
+
+=== Filter Ordering Best Practices
+
+1. **OpenTelemetry filters should be among the first filters** to ensure proper context establishment
+2. **Use non-offloading filters** (`appendNonOffloadingServiceFilter`) for OpenTelemetry filters to maintain context
+3. **Request draining must come after OpenTelemetry** on the server side
+4. **Exception mapping should come after OpenTelemetry** to ensure trace status reflects actual response codes
+5. **Lifecycle observers should come after OpenTelemetry** to see correct span information
+
+== Context Propagation
+
+OpenTelemetry context is automatically propagated through multiple mechanisms to ensure traces are correlated correctly across service boundaries and async operations.
+
+=== Header Propagation
+
+OpenTelemetry context is automatically injected into and extracted from headers using the standard OpenTelemetry propagation format:
+
+* **W3C Trace Context** (`traceparent`, `tracestate` headers)
+* **B3 Propagation** (if configured)
+* **Custom propagators** (if configured in the OpenTelemetry SDK)
+
+[source,java]
+----
+// Context is automatically propagated via headers
+HttpResponse response = client.request(client.get("/api/endpoint"));
+// The server will receive trace context via HTTP headers
+----
+
+=== Async Context Integration
+
+ServiceTalk's async context system ensures OpenTelemetry context is maintained across:
+
+* **Thread boundaries** during async operations
+* **Publisher/Subscriber chains** in reactive streams
+* **Executor transitions** when work is offloaded
+* **Filter chains** where context must be preserved
+
+This integration is provided by the `servicetalk-opentelemetry-asynccontext` module, which implements the `CapturedContextProvider` interface.
+
+=== Context Scope Management
+
+OpenTelemetry spans are automatically activated and deactivated at appropriate points:
+
+[source,java]
+----
+// Client side: span is active during request processing
+client.request(client.get("/api"))
+    .beforeOnSuccess(response -> {
+        // Current span is still active here
+        Span currentSpan = Span.current();
+        currentSpan.setStatus(StatusCode.OK);
```

Comment:
Is it a legic example to setStatus manually? Should we use a custom attribute as an example instead?

Reply:
I believe it is a legit example. iirc, the instrumentation will only set errors, but not OK, because it considers it a business logic concern to know if it was truly OK and not just a successful RPC but a bad result that was returned.

Reply:
That convention is defined here: https://github.com/open-telemetry/opentelemetry-specification/blob/v1.48.0/specification/trace/api.md#set-status
With the most relevant quote being:
```
Generally, Instrumentation Libraries SHOULD NOT set the status code to Ok, unless explicitly configured to do so. Instrumentation Libraries SHOULD leave the status code as Unset unless there is an error, as described above.

Application developers and Operators may set the status code to Ok.
```